### PR TITLE
Add azure-nextgen-go-static-website example

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

File: azure-nextgen-go-static-website/go.mod [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module github.com/pulumi/examples/azure-go-static-website
+
+go 1.15
+
+require (
+	github.com/pulumi/pulumi-azure-nextgen/sdk v0.6.2-0.20210218161332-bbda61784b00
```
Comment (mikhailshilkov) [inline azure-nextgen-go-static-website/go.mod:6]: Why the latest alpha build?

File: azure-nextgen-go-static-website/main.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		})
+		if err != nil {
+			return err
+		}
+		_, err = storageDefault.NewBlob(ctx, "notfound_html", &storageDefault.BlobArgs{
+			BlobName:          pulumi.String("404.html"),
+			ResourceGroupName: resourceGroup.Name,
+			AccountName:       storageAccount.Name,
+			ContainerName:     staticWebsite.ContainerName,
+			Type:              &blobTypeBlock,
```
Comment (mikhailshilkov) [inline azure-nextgen-go-static-website/main.go:160]: If we want, we can remove a bunch of optional parameters (like `Type` here). Should I list them all?

Comment (lblackstone) [reply azure-nextgen-go-static-website/main.go:160]: Yeah, that sounds good. I've just been copying the TS example, so open to suggestions on simplifying.

Comment (mikhailshilkov) [reply azure-nextgen-go-static-website/main.go:160]: Let's merge it and I'll take a pass when we need to change to azure-native

Comment (lblackstone) [reply azure-nextgen-go-static-website/main.go:160]: > Let's merge it and I'll take a pass when we need to change to azure-native

I think this example needs to wait on the next provider release. I'm currently getting the following error: `error: Resource type azure-nextgen:storage:StorageAccountStaticWebsite not found` and it seems like some of the APIs aren't available in the current latest release.

Comment (mikhailshilkov) [reply azure-nextgen-go-static-website/main.go:160]: 0.6.1 should have all that's needed... Other languages are fine on it, right?

Comment (lblackstone) [reply azure-nextgen-go-static-website/main.go:160]: Ok, got it working. I think I was missing a `latest` tag before.

**Pull request merged**